File: Java/Java/ExceptionDate.java
import java.util.*;

class InvalidDate extends Exception{
    
    public String toString(){
        return "InvalidDateException";
    }
}

class ddmmyy{
    
    int date,month,year;
    Scanner s = new Scanner(System.in);
    
    void Accept(){
             System.out.print("Enter Date : ");
             date = s.nextInt();
             System.out.print("Enter Month : ");
             month = s.nextInt();
             System.out.print("Enter Year : ");
             year = s.nextInt();
        }
    
    void Display(){
        System.out.print(date+"/"+month+"/"+year);
    }
    
}

public class ExceptionDate {
    
    public static void main(String[] args) {
        
        ddmmyy d = new ddmmyy();
        
        try{
            
            d.Accept();
            
            if(d.month < 1 || d.month > 12){
                throw new InvalidDate();
            }
            
            switch(d.month){
                case 1:
                case 3:
                case 5:
                case 7:
                case 8:
                case 10:
                case 12:
                if(d.date < 1 || d.date>31){
                    throw new InvalidDate();
                }
                break;
                case 4:
                case 6:
                case 9:
                case 11:
                if(d.date < 1 || d.date > 30){
                    throw new InvalidDate();
                }
                break;
                case 2:
                int max;
                if(d.year%4==0 || d.year%400==0){
                    max = 29;
                }
                else{
                    max = 28;
                }
                
                if(d.date > max){
                    throw new InvalidDate();
                }
                break;
            }
            
            d.Display();
        }
        catch (InvalidDate e){
            System.out.print(e);
        }
        
    }
    
}

File: Python/Exception Date.py

class InvalidDate(Exception):
    pass
    
class Date:
    
    def Accept(self):
      try: 
        self.date = int(input("Enter the Date : "))
        self.month = int(input("Enter the Month : "))
        self.year = int(input("Enter the Year : "))
        self.Conditions()
      except ValueError:
          raise InvalidDate("Invalid Date Format! Enter Must Integer")
    
    def Conditions(self):
        
        if (self.month < 1 or self.month > 12 ):
          raise InvalidDate("InvalidMonth! Plz Check Your Month")
       
        elif (self.date < 1):
            raise InvalidDate("Date is Not Less than 1")
        
        elif (self.month in [1,3,5,7,8,10,12] and self.date > 31):
            raise InvalidDate("InvalidDate! Date Must Between 1 to 31")
       
        elif (self.month in [4,6,9,11] and self.date > 30):
            raise InvalidDate("InvalidDate! Date Must Between 1 to 30")
       
        if(self.month == 2 ):
            
                if(self.year % 4 == 0 or self.year % 400 ==0):
                    dd = 29
                else:
                    dd = 28  
                      
                if(self.date > dd):
                     raise InvalidDate("InvalidDate!")
                     
    def Display(self):
        print(f"{self.date}/{self.month}/{self.year}")             
                
d = Date()  

try:
    d.Accept()  
    d.Display() 
except InvalidDate as e:
    print(e) 